### PR TITLE
Fix issues with julia10

### PR DESCRIPTION
**Pull request by vitornesello**
@mlubin These should fix https://github.com/JuliaOpt/Gurobi.jl/pull/157 but it will not work with Julia 0.6 because the function `spdiagm` was changed in 1.0

### CODE REVIEW

**Comment by mlubin** [comment]
Is it possible to use `Compat` to keep the code working on 0.6?

**Comment by vitornesello** [comment]
The Travis tests worked for 0.6, but I don't know why. Maybe it did not run that particular line. 

> Is it possible to use `Compat` to keep the code working on 0.6?

I will check it tomorrow

**Comment by mlubin** [comment]
> The Travis tests worked for 0.6, but I don't know why. Maybe it did not run that particular line.

Yep, these lines aren't covered by tests, so we have to be extra careful.

> I will check it tomorrow

Thanks!

**Comment by joaquimg** [comment]
This is nice, but we still need to disallow fails on travis, which currently contains:
```
matrix:
    allow_failures:
         - julia: 1.0
```
We might need to fix GLPK also.

**Comment by vitornesello** [comment]
> We might need to fix GLPK also.

Yes, tests fail because of the `GLPKMathProgInterface`

**Comment by mlubin** [comment]
This PR doesn't need to be blocked by GLPKMathProgInterface and making travis pass on 1.0 (we already have #224 open). OTOH we can't merge if the tests are failing, so please revert the change to disallow failures.

File: src/SolverInterface/lpqp_to_conic.jl [hunk 1]
```diff
@@ -154,9 +154,11 @@ function loadproblem!(m::LPQPtoConicBridge, c, A, b, constr_cones, var_cones)
         # rsoc has x'x <= 2pq
         # quadratic form is x'x <= pq
         diagvec = ones(length(rsocconstr_idx))
-        diagvec[rsoc_start_idx] = 1/sqrt(2)
+        for rsoc_idx in rsoc_start_idx
+            diagvec[rsoc_idx] = 1/sqrt(2)
+        end
         Alin = [ Alin spzeros(size(Alin,1),length(rsocconstr_idx))
-        [m.Arsoc spzeros(size(m.Arsoc,1),length(socconstr_idx))] spdiagm(diagvec) ]
+        [m.Arsoc spzeros(size(m.Arsoc,1),length(socconstr_idx))] @compat spdiagm(0 => diagvec) ]
```
Comment (mlubin) [inline src/SolverInterface/lpqp_to_conic.jl]: I don't think the compat macro is needed here. `SparseArrays.spdiagm(0 => diagvec)` should do the trick.

File: src/SolverInterface/lpqp_to_conic.jl [hunk 2]
```diff
@@ -154,9 +154,11 @@ function loadproblem!(m::LPQPtoConicBridge, c, A, b, constr_cones, var_cones)
         # rsoc has x'x <= 2pq
         # quadratic form is x'x <= pq
         diagvec = ones(length(rsocconstr_idx))
-        diagvec[rsoc_start_idx] = 1/sqrt(2)
+        for rsoc_idx in rsoc_start_idx
```
Comment (mlubin) [inline src/SolverInterface/lpqp_to_conic.jl]: `diagvec[rsoc_start_idx] .= 1/sqrt(2)`?

File: src/SolverInterface/lpqp_to_conic.jl (hunk 1 above)
Comment (vitornesello) [reply src/SolverInterface/lpqp_to_conic.jl]: I tried this syntax in 0.6, it returns `MethodError: no method matching spdiagm(::Pair{Int64,Array{Int64,1}})`

Comment (vitornesello) [reply src/SolverInterface/lpqp_to_conic.jl]: Sorry, I will fix it like you suggested in https://github.com/JuliaOpt/GLPKMathProgInterface.jl/pull/52

Comment (mlubin) [reply src/SolverInterface/lpqp_to_conic.jl]:
```julia
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.3 (2018-05-28 20:20 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

julia> import Compat.SparseArrays

julia> SparseArrays.spdiagm(0 => [1, 2])
2×2 SparseMatrixCSC{Int64,Int64} with 2 stored entries:
  [1, 1]  =  1
  [2, 2]  =  2
```

**Pull request merged**